sdk/search/Azure.Search.Documents/tests/Utilities/SearchResources.cs

```diff
@@ -455,40 +455,58 @@ private async Task<SearchResources> CreateHotelsBlobContainerAsync(bool populate
 
                 if (populate)
                 {
-                    object[] hotels = isSample ? SearchResourcesSample.TestDocumentsForSample : TestDocuments;
-                    List<Task> tasks = new List<Task>(hotels.Length);
-
-                    foreach (object obj in hotels)
+                    if (isSample)
                     {
-                        Task task = Task.Run(async () =>
+                        Samples.Hotel[] hotels = SearchResourcesSample.TestDocumentsForSample;
+                        List<Task> tasks = new List<Task>(hotels.Length);
+
+                        foreach (Samples.Hotel hotel in hotels)
                         {
```

Comment:
[nit] You could move the common code like tasks and waiting on them outside of if\else and share it.